### PR TITLE
Fix custom element initialization

### LINKED ISSUE #342: Buggy custom element behavior when script is included before element is used
**Bug** 

Including the `<script>` tag for the dojo button element before the element is referenced, as in this snippet:

```html
<!DOCTYPE html>
<html>
<head>
	<link rel="stylesheet" href="output/dist/button-6.0.0-pre.css" />
	<script src="output/dist/button-6.0.0-pre.js"></script>
</head>
<body>
<dojo-button>Hello</dojo-button>
</body>
</html>
```

Will cause the text Hello to not be inserted correctly as a child element.

image

Moving the script tag to the bottom, as in:

```html
<!DOCTYPE html>
<html>
<head>
	<link rel="stylesheet" href="output/dist/button-6.0.0-pre.css" />
</head>
<body>
<dojo-button>Hello</dojo-button>
<script src="output/dist/button-6.0.0-pre.js"></script>
</body>
</html>
```

Results in the proper DOM structure.

image

Package Version: 6.0.0-alpha.1

**Code**

I just built the @dojo/widgets custom elements and put the previously posted html file(s) in the root.

**Expected behavior:**

Custom elements will be initialized correctly regardless of whether or not their script has been loaded first.

**Actual behavior:**

Custom elements are only initialized correctly if their script is loaded after the element has been used.

### PR DESCRIPTION
**Pull request by maier49**
**Type:** bug

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [x] Unit or Functional tests are included in the PR

**Description:**
It's evidently not guaranteed that a custom element's children will be available when it is connected ([see discussion here](https://github.com/w3c/webcomponents/issues/551)), but our `connectedCallback` was setup as if it was. I've setup a mutation observer so that the element will wait till the document is loaded  or an element after it in the HTML hierarchy is present to perform setup logic. 

Resolves #342

### CODE REVIEW

**Comment by agubler** [comment]
@maier49 FYI, this seems to be failing CI.

File: tests/core/unit/registerCustomElement.ts [hunk 1]
```diff
@@ -349,4 +372,35 @@ describe('registerCustomElement', () => {
...
+		Object.defineProperty(document, 'readyState', {
+			configurable: true,
+			get() {
+				return 'complete';
+			}
+		});
+		child = document.createTextNode('bar');
+		element!.appendChild(child);
+
+		waitFor(
```
Comment (agubler) [inline tests/core/unit/registerCustomElement.ts]: curious, do you need to return this to ensure that the test waits for the promise to be resolved?

Comment (maier49) [reply tests/core/unit/registerCustomElement.ts]: Yes 🤦‍♂ . This is actually broken in IE. Looking into it

Comment (maier49) [reply tests/core/unit/registerCustomElement.ts]: @agubler Ok it was still the test that was broken I just didn't fix it properly. Should be good now (pending CI)

**Comment by maier49** [comment]
@agubler A problem with the approach I was using is that if no children are appended it might never call the ready callback. I've updated it to use `setTimeout`.

**Comment by maier49** [comment]
@agubler Testing against the v5 custom elements showcase everything worked properly out of the box. But there were was an issue with the setup being called too many times when I put the element scripts in the head. Have that fixed now, here's the showcase with the scripts in the head:
https://dist-ytwugedbhm.now.sh/

**Pull request merged**